Sources/Benchmarks/Pose2SLAM.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Benchmarks Pose2SLAM solutions.
+
+import Benchmark
+import SwiftFusion
+
+let pose2SLAM = BenchmarkSuite(name: "Pose2SLAM") { suite in
+
+  let intelDataset = try! G2OFactorGraph(fromG2O: try! cachedDataset("input_INTEL_g2o.txt"))
+  check(intelDataset.graph.error(intelDataset.initialGuess), near: 73565.64, accuracy: 1e-2)
+
+  // Uses `NonlinearFactorGraph` on the Intel dataset.
+  // The solvers are configured to run for a constant number of steps.
+  // The nonlinear solver is 5 iterations of Gauss-Newton.
+  // The linear solver is 100 iterations of CGLS.
+  suite.benchmark(
+    "NonlinearFactorGraph, Intel, 5 Gauss-Newton steps, 100 CGLS steps",
+    settings: .iterations(1)
```

Comment:
Results are going to be quite noisy if you only run a single iteration. Is it not practical to run it for longer?

Reply:
This is benchmarking a whole solver, and it's pretty slow right now (~12 seconds).

Reply:
Thanks for the clarification.